### PR TITLE
Add no ic feature extraction

### PR DESCRIPTION
**Pull request by roshankern**
This PR is ready for review!

In this PR, we extract all single-cell features without performing `PyBasic` illumination correction on all images. The resulting datasets (with/without illumination correction) are specified with `dataset_type` (`ic` or `no_ic`).

### CODE REVIEW

File: 1.idr_streams/README.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ In [streams/](streams/) we initialize and run `idrstream` for the training, nega
 The `batch_size` parameter tells `idrstream` how many frames to process in one batch.
 We set this to `10` for MitoCheck data, meaning that the features for cells in 10 images (unique plate/well/frame combination) are extracted in each batch.
 
+We set `perform_illumination_correction` equal to `True` or `False` depending on the datset type we are extracting (`ic` or `no_ic`).
```
Comment (gwaybio) [inline 1.idr_streams/README.md:16]: Can you add some details about what we're testing here? Why is it important to rerun everything with and without IC?

File: 1.idr_streams/streams/cp_streams.py [hunk 2]
```diff
@@ -60,10 +65,11 @@
...
+    stream.init_downloader(aspera_path, aspera_key_path, screens_path, idr_index_name)
 
     # init preprocessor
-    stream.init_preprocessor(fiji_path)
+    perform_illumination_correction = True if dataset_type == "ic" else False
+    stream.init_preprocessor(fiji_path, perform_illumination_correction)
 
     # run cp IDR_stream!
-    stream.run_cp_stream(data_to_process, batch_size=3, start_batch=0, batch_nums=[0])
+    stream.run_cp_stream(data_to_process, batch_size=10, start_batch=0)
```
Comment (gwaybio) [inline 1.idr_streams/streams/cp_streams.py:75]: cool to see this variable update!

File: 2.format_training_data/scripts/nbconverted/format_training_data.py [hunk 3]
```diff
@@ -16,56 +16,42 @@
...
+for dataset_type in dataset_types:
+    print(f"Compiling data for dataset type {dataset_type}")
+
+    # Load training data
+    training_data_features_path = pathlib.Path(f"../1.idr_streams/extracted_features/training_data__{dataset_type}/merged_features/")
+    training_data = compile_mitocheck_batch_data(training_data_features_path)
+    print(f"Training data has {training_data.shape}")
+    
+    # Find cells with MitoCheck data
+    # Cells may not have feature data found because the specific well is not hosted on IDr (failure to pass QC) or because of differences in Mitocheck/DeepProfiler segmentation
```
Comment (gwaybio) [inline 2.format_training_data/scripts/nbconverted/format_training_data.py:44]: love this documentation!

File: 2.format_training_data/scripts/nbconverted/format_training_data.py [hunk 4]
```diff
@@ -16,56 +16,42 @@
...
+    features_samples_path = pathlib.Path("../mitocheck_metadata/features.samples.txt")
+    training_cells = get_labeled_cells(training_data, features_samples_path, "DP__Object_Outline")
+    print(f"Shape of labeled_cells: {training_cells.shape}")
+    
+    # Replace `Shape1` and `Shape3` with their respective classes
+    # See https://github.com/WayScience/mitocheck_data/issues/16 for more details
```
Comment (gwaybio) [inline 2.format_training_data/scripts/nbconverted/format_training_data.py:50]: this is great too!

File: 4.analyze_data/results/single_cell_class_counts.csv [hunk 5]
```diff
@@ -1,17 +1,33 @@
...
-7,SmallIrregular,164
-8,Hole,114
-9,Elongated,110
-10,ADCCM,95
-11,Anaphase,84
-12,Large,79
-13,Grape,74
-14,Metaphase,74
-15,Folded,54
+,Mitocheck_Phenotypic_Class,Single_Cell_Counts,Dataset_Type
```
Comment (gwaybio) [inline 4.analyze_data/results/single_cell_class_counts.csv:1]: this  is very interesting to see the differences!

**Review by gwaybio — APPROVED** [review]
:tada: looks great @roshankern !

File: 1.idr_streams/streams/cp_streams.py (hunk 2 above)
Comment (roshankern) [reply 1.idr_streams/streams/cp_streams.py:75]: Yep, going through each part of this repo was definitely good for fixing some small mistakes that made it through the PRs :)

**Pull request merged**